testutil/retry/retry.go

```diff
@@ -0,0 +1,125 @@
+// Package retry provides a generic retry mechanism
+// which can be used in tests.
+package retry
+
+import "time"
+
+const (
+	// Timeout is the default time span for which an operation
+	// should be retried.
+	Timeout = time.Second
+
+	// Wait is the time between two retry attempts.
+	Wait = 25 * time.Millisecond
+)
+
+// Retryer provides an interface for retrying an operation
+// repeatedly until it either succeeds or times out. The
+// Failer will be called when on timeout.
+//
+// Retryer does not accept a callback function to execute
+// the tests to keep the file:line information generated
+// by test output correct.
+//
+//   func TestX(t *testing.T) {
+//       for r := retry.For{}{}; r.Next(); {
+// 		     if err := f(); err != nil {
+// 			     t.Log("f: ", err)
+// 			     continue
+// 		     }
+// 		     break
+// 	     }
+//   }
+type Retryer interface {
+	// Next returns true if the operation can be retried.
+	// If not, it will call t.FailNow() and return false.
+	Next(t Failer) bool
+
+	// Reset configures the retryer for re-use.
+	Reset()
+}
+
+// Failer is an interface compatible with *testing.T.
+type Failer interface {
+	FailNow()
+}
+
+// R returns a Timer with default configuration.
+func R() *Timer {
```

Comment:
I agree and I've struggled with the default values in general since they are arbitrary. I've dropped the `Timeout` and `Wait` defaults since there is no need to expose them and created more explicit names `OneSec` and `ThreeTimes` for the convenience implementations.